File: bm2.rb
require 'red_black_tree'

def bm(arr)
  start = Time.now

  tree = RedBlackTree.new
  arr.each do |x|
    tree.add(x)
  end

  return Time.now - start
end

N = (ARGV[0] || 5).to_i

arr = []
$stdin.each_line {|line| arr << line.to_i }

N.times do
  puts bm(arr)
end

File: bm2.py
import sys
import time
import random
from red_black_tree import RedBlackTree
import sys

def bm(arr):
  start = time.time()

  tree = RedBlackTree()
  for x in arr:
    tree.add(x)

  return time.time() - start

if len(sys.argv) > 1:
  n = int(sys.argv[1])
else:
  n = 5

arr = []
for line in sys.stdin:
  arr.append(int(line))

for i in range(0, n):
  print("%02f" % bm(arr))

